marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/preview-matching-activity-lib.xqy

```diff
@@ -158,11 +156,47 @@ declare function pma:preview-matching-activity(
         pma:match-against-source-query-docs($uris, $options, $source-query, $previous-count)
       )
   let $all-results := fn:subsequence(($results-within-uris, $results-against-source-query-docs), 1, $PMA-MAX-RESULTS)
+  let $all-uris := fn:distinct-values(($all-results ! ( json:array-values(map:get(., "uris")))))
+  let $entity-type := $options/targetEntityType
+  let $primary-keys := pma:primary-keys-for-uris($entity-type, $all-uris)
   let $_ :=
   (
     map:put($obj, "sampleSize", $sample-size),
+    map:put($obj, "primaryKeys", $primary-keys),
     map:put($obj, "uris", json:to-array($uris)),
     map:put($obj, "actionPreview", json:to-array($all-results))
   )
   return xdmp:to-json($obj)/node()
 };
+
+declare function pma:primary-keys-for-uris(
```

Comment:
This seems generic to entity instances - i.e. "I have N entity instance URIs, and I know the entity type; I want identifiers for all of them. 

That seems like a candidate for adding to entity-lib - e.g. "getEntityIdentifiers(uris, entityType)" - but that would have the downside here of XQuery having to call JS. 

Thoughts on that sort of function living in hub-entities.xqy, which at least is in XQuery?

Reply:
That makes sense. I'll move it to hub-entities.xqy and provide also add a JS function in entity-lib that calls the XQuery.

Should I make the signature `find-entity-identifiers($uris, $entity-type)/findEntityIdentifiers(uris, entityType) ` since there is a SPARQL query?

Reply:
Yes, "find" is a good choice.

Reply:
@rjrudin Okay. I've updated the PR with the function moved and added additional tests specifically for the common JS function.